### PR TITLE
Add unit tests for org.zaproxy.zap.utils.ByteBuilder

### PR DESCRIPTION
**Pull request by Braavos6**
I've analysed your codebase and noticed that `org.zaproxy.zap.utils.ByteBuilder` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    public void testEnsureCapacity() {
+        ByteBuilder byteBuilder = new ByteBuilder(5);
+        Assert.assertEquals(5, byteBuilder.capacity());
+
+        byteBuilder.ensureCapacity(20);
+        Assert.assertEquals(20, byteBuilder.capacity());
+    }
+
+    @Test
+    public void testEnsureSize() {
```
Comment (petersear) [inline zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java]: There is no ensureSize() method so this should probably be testSize()?

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    }
+
+    @Test
+    public void testAppendCharArray2() {
+        ByteBuilder byteBuilder = new ByteBuilder(new byte[]{4, 4});
+        byteBuilder.append(new char[]{'1', '2', '3', '4', '5', '6'}, 1, 3);
+
+        Assert.assertArrayEquals(new byte[]{4, 4, 50, 51, 52},
+                byteBuilder.toByteArray());
+        thrown.expect(ArrayIndexOutOfBoundsException.class);
```
Comment (petersear) [inline zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java:107]: Insert blank line between assert and exception sections.

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        byteBuilder.append(2);
+
+        Assert.assertArrayEquals(new byte[]{4, 4, 0, 0, 0, 2},
+                byteBuilder.toByteArray());
+    }
+
+    @Test
+    public void testAppendLong() {
+        ByteBuilder byteBuilder = new ByteBuilder(new byte[]{4, 4});
+        byteBuilder.append(1234567890123456789l);
```
Comment (petersear) [inline zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java]: It's common to use `L` to denote `long` and avoid confusion with `1`

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        byteBuilder.append(new ByteBuilder(new byte[]{5, 5}));
+
+        Assert.assertArrayEquals(new byte[]{4, 4, 5, 5},
+                byteBuilder.toByteArray());
+    }
+
+    @Test
+    public void testAppendSpecial1() {
+        ByteBuilder byteBuilder = new ByteBuilder(new byte[]{4, 4});
+        byteBuilder.appendSpecial(1234567890123456789l, 3, true);
```
Comment (petersear) [inline zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java]: `l` -> `L`

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+        Assert.assertArrayEquals(new byte[]{4, 4, 105, -127, 21},
+                byteBuilder.toByteArray());
+    }
+
+    @Test
+    public void testAppendSpecial2() {
+        ByteBuilder byteBuilder = new ByteBuilder(new byte[]{4, 4});
+        byteBuilder.appendSpecial(-1234567890123456789l, 3, true);
```
Comment (petersear) [inline zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java]: `l` -> `L`

**Review by petersear — CHANGES_REQUESTED** [review]

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java (hunk 1 above)
Comment (Braavos6) [reply zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java]: I was going for creativity there :p 
Done

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java (hunk 2 above)
Comment (Braavos6) [reply zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java:107]: Done

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java (hunk 3 above)
Comment (Braavos6) [reply zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java]: Done

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java (hunk 4 above)
Comment (Braavos6) [reply zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java]: Done

File: zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java (hunk 5 above)
Comment (Braavos6) [reply zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderTest.java]: Done

**Review by petersear — APPROVED** [review]

**Review by freddytuxworth — APPROVED** [review]

**Pull request closed without merging**